extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultPKIManager.java

```diff
@@ -0,0 +1,658 @@
+    }
+
+    private String getToken() {
+        return vaultAuthManager.getClientToken();
+    }
+
+    @Override
+    public CertificateData.PEM getCertificateAuthority() {
+        VaultPKICertificateResult internalResult = vaultInternalPKISecretEngine.getCertificate(getToken(), mount, "ca");
+        checkDataValid(internalResult);
+
+        return new CertificateData.PEM(internalResult.data.certificate);
+    }
+
+    @Override
+    public void configCertificateAuthority(String pemBundle) {
+        VaultPKIConfigCABody body = new VaultPKIConfigCABody();
+        body.pemBundle = pemBundle;
+        vaultInternalPKISecretEngine.configCertificateAuthority(getToken(), mount, body);
+    }
+
+    @Override
+    public void configURLs(ConfigURLsOptions options) {
+        VaultPKIConfigURLsData body = new VaultPKIConfigURLsData();
+        body.issuingCertificates = options.issuingCertificates;
+        body.crlDistributionPoints = options.crlDistributionPoints;
+        body.ocspServers = options.ocspServers;
+
+        vaultInternalPKISecretEngine.configURLs(getToken(), mount, body);
+    }
+
+    @Override
+    public ConfigURLsOptions readURLsConfig() {
+        VaultPKIConfigURLsResult internalResult = vaultInternalPKISecretEngine.readURLs(getToken(), mount);
+        checkDataValid(internalResult);
+
+        VaultPKIConfigURLsData internalResultData = internalResult.data;
+
+        ConfigURLsOptions result = new ConfigURLsOptions();
+        result.issuingCertificates = internalResultData.issuingCertificates;
+        result.crlDistributionPoints = internalResultData.crlDistributionPoints;
+        result.ocspServers = internalResultData.ocspServers;
+        return result;
+    }
+
+    @Override
+    public void configCRL(ConfigCRLOptions options) {
+        VaultPKIConfigCRLData body = new VaultPKIConfigCRLData();
+        body.expiry = options.expiry;
+        body.disable = options.disable;
+
+        vaultInternalPKISecretEngine.configCRL(getToken(), mount, body);
+    }
+
+    @Override
+    public ConfigCRLOptions readCRLConfig() {
+        VaultPKIConfigCRLResult internalResult = vaultInternalPKISecretEngine.readCRL(getToken(), mount);
+        checkDataValid(internalResult);
+
+        VaultPKIConfigCRLData internalResultData = internalResult.data;
+
+        ConfigCRLOptions result = new ConfigCRLOptions();
+        result.expiry = internalResultData.expiry;
+        result.disable = internalResultData.disable;
+        return result;
+    }
+
+    @Override
+    public String getCertificateAuthorityChain() {
+        VaultPKICertificateResult internalResult = vaultInternalPKISecretEngine.getCertificate(getToken(), mount, "ca_chain");
+        checkDataValid(internalResult);
+
+        return internalResult.data.certificate;
+    }
+
+    @Override
+    public String getCertificateRevocationList() {
+        VaultPKICertificateResult internalResult = vaultInternalPKISecretEngine.getCertificate(getToken(), mount, "crl");
```

Comment:
I am not using `/pki/crl` endpoint because it returns _raw_ binary data; which isn't support by `VaultClient`. There is no JSON wrapper. Instead I am using `/pki/cert/crl` which returns data in PEM.

Reply:
I've switched to the `pki/crl` endpoint and added a method `getCertificateRevocationList(DataFormat)` in addition to `getCertificateRevocationList()`; they both return `CRLData` objects.